### PR TITLE
Add support for setting ignore fields for each resource key

### PR DESCRIPTION
**Pull request by funera1**
**What this PR does / why we need it**:

It didn’t distinguish fields by resource key to ignore drift detection. So, this PR resolves the problem.

Input form is `apiVersion:kind:namespace:name#filedPath`.

For example,
```
…
driftDetection: 
    ignoreFields: 
        - apps/v1:Deployment:default:simple#spec.replicas 
        - apps/v1:Deployment:default:simple#spec.template.spec.containers.0.args
```

You may find it annoying to have to specify the Key, but we plan to add a future feature that will allow you to easily copy the path of the field you want to ignore from the web ui

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/driftdetector/kubernetes/detector.go [hunk 1]
```diff
@@ -193,9 +193,17 @@ func (d *detector) checkApplication(ctx context.Context, app *model.Application,
 		return err
 	}
 
-	ignoreFields := make([]string, 0)
+	ignoreConfig := make(map[string][]string, 0)
```
Comment (khanhtc1202) [inline pkg/app/piped/driftdetector/kubernetes/detector.go:196]: nits, can we rename it ad ignoreConfig 👀

File: pkg/diff/diff.go [hunk 2]
```diff
@@ -59,14 +60,14 @@ func WithCompareNumberAndNumericString() Option {
 }
 
 // WithIgnoredPaths configures ignored fields.
-func WithIgnoredPaths(paths []string) Option {
+func WithIgnoreConfig(config map[string][]string) Option {
```
Comment (khanhtc1202) [inline pkg/diff/diff.go:63]: change the comment on above this function header, the name is not matched

File: pkg/app/piped/driftdetector/kubernetes/detector.go [hunk 3]
```diff
@@ -193,9 +193,17 @@ func (d *detector) checkApplication(ctx context.Context, app *model.Application,
...
 	if ddCfg != nil {
-		ignoreFields = ddCfg.IgnoreFields
+		for _, ignoreField := range ddCfg.IgnoreFields {
+			// ignoreField is 'apiVersion:kind:namespace:name#fieldPath'
```
Comment (khanhtc1202) [inline pkg/app/piped/driftdetector/kubernetes/detector.go:199]: Should add this line to the config/application as well to mention what we expect as driftDetection.ignoreFields

Comment (funera1) [reply pkg/app/piped/driftdetector/kubernetes/detector.go:199]: I changed it. How about that?

File: pkg/diff/diff.go (hunk 2 above)
Comment (funera1) [reply pkg/diff/diff.go:63]: :+1:

File: pkg/app/piped/driftdetector/kubernetes/detector.go (hunk 3 above)
Comment (khanhtc1202) [reply pkg/app/piped/driftdetector/kubernetes/detector.go:199]: 👍

**Review by khanhtc1202 — APPROVED** [review]
Nice work, thank you 🙌

**Pull request merged**